Allow to inject compiler pragmas in generated file (#19906)
coq/coq: doc/sphinx/addendum/extraction.rst

   This :term:`option` provides a comment that is
   included at the beginning of the output files.

.. opt:: Extraction File Pragma @string

>    This :term:`option` provides a compiler pragma that is
>    included at the beginning of the output files.
>    ``Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".``

Reviewer: ```suggestion
   Specifies a compiler pragma that is
   included at the beginning of the output files.
   For example, ``Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".``
```

Based on your Haskell example, please explain that, IINM, the Pragma is embedded in a comment in the output.

We try to avoid using repetitive boilerplate such as "This option provides", though that still appears in many places.  The header in the generated doc will already label it as an option.

Not sure the example adds much.  Would be better if you very briefly said what `OPTIONS_GHS -cpp -XMagicHash` does.

Author: The example has been commented.

   This :term:`option` provides a comment that is
   included at the beginning of the output files.

.. opt:: Extraction File Pragma @string

   Specifies a compiler `pragma
   <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html#options-ghc-pragma>`_
   that is included at the beginning of the output files.

>    For example:
> 
>    .. coqtop:: in
> 
>       Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".
> 
>    The pragma enables C preprocessor and `MagicHash
>    <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/magic_hash.html#the-magic-hash>`_
>    language extension.
> 
>    .. note::
> 
>       The option has sense only for extraction to `Haskell or JSON <#grammar-token-language>`_.
> 

Reviewer: ```suggestion
   For example, for a Haskell extraction, `Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".`
   enables the C preprocessor and the `MagicHash
   <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/magic_hash.html#the-magic-hash>`_
   language extension.
```

Not much benefit in calling coqtop for a single line that doesn't show any output.  And I think easier to read as a single sentence rather than with a lot of formatting.  Just a thought/suggestion.
